lib/ansible/modules/cloud/cloudstack/cs_vlan_ip_range.py

```diff
@@ -0,0 +1,327 @@
+
+                params = {
+                    'startip': self.module.params.get('start_ip'),
+                    'endip': self.get_or_fallback('end_ip', 'start_ip'),
+                }
+
+                for ipr in ip_range_list:
+                    if params['startip'] == ipr['startip'] and params['endip'] == ipr['endip']:
+                        self.ip_range = ipr
+
+        return self.ip_range
+
+    def present_vlan_ip_range(self):
+        ip_range = self.get_vlan_ip_range()
+
+        if not ip_range:
+            ip_range = self.create_vlan_ip_range()
+
+        return ip_range
+
+    def create_vlan_ip_range(self):
+        self.result['changed'] = True
+
+        vlan = self.module.params.get('vlan')
+
+        args = {
+            'zoneid': self.get_zone(key='id'),
+            'projectid': self.get_project(key='id'),
+            'account': self.get_account(key='name'),
+            'domainid': self.get_domain(key='id'),
+            'startip': self.module.params.get('start_ip'),
+            'endip': self.get_or_fallback('end_ip', 'start_ip'),
+            'netmask': self.module.params.get('netmask'),
+            'gateway': self.module.params.get('gateway'),
+            'startipv6': self.module.params.get('start_ipv6'),
+            'endipv6': self.get_or_fallback('end_ipv6', 'start_ipv6'),
+            'ip6gateway': self.module.params.get('gateway_ipv6'),
+            'vlan': self.get_network(key='vlan') if not vlan else vlan,
+            'networkid': self.get_network(key='id'),
+            'forvirtualnetwork': self.module.params.get('for_virtual_network'),
+        }
+
+        if not self.module.check_mode:
+            res = self.query_api('createVlanIpRange', **args)
+
+            self.ip_range = res['vlan']
+
+        return self.ip_range
+
+    def absent_vlan_ip_range(self):
+        ip_range = self.get_vlan_ip_range()
+
+        if ip_range:
+            self.result['changed'] = True
+
+            args = {
+                'id': ip_range['id'],
+            }
+
+            if not self.module.check_mode:
+                self.query_api('deleteVlanIpRange', **args)
+
+        return ip_range
+
+
+def main():
+    argument_spec = cs_argument_spec()
+    argument_spec.update(dict(
+        network=dict(type='str', required=True),
+        zone=dict(type='str'),
+        start_ip=dict(type='str', required=True),
+        end_ip=dict(type='str'),
+        gateway=dict(type='str', required=True),
+        netmask=dict(type='str', required=True),
+        start_ipv6=dict(type='str'),
+        end_ipv6=dict(type='str'),
+        gateway_ipv6=dict(type='str'),
+        vlan=dict(type='str'),
+        state=dict(choices=['present', 'absent'], default='present'),
+        domain=dict(type='str'),
+        account=dict(type='str'),
+        project=dict(type='str'),
+        for_virtual_network=dict(type='bool', default=False),
+        for_systemvms=dict(type='bool', default=False),
```

Comment:
this param seems unused. remove?

Reply:
It must be useful in some cases, but as I don't know how to test it, I will remove it for the moment.